src/components/form/templates/grid-layout.ts

```diff
@@ -0,0 +1,92 @@
+import React from 'react';
+import { FormLayoutOptions, GridLayoutOptions } from '../form.types';
+
+const MAX_COLUMNS = 5;
+const MAX_COLUMN_WIDTH = 15;
+const REM_TO_PX = 16;
+const COLUMN_COUNT = '--number-of-columns';
+
+export class GridLayout extends React.Component<LayoutProps, LayoutState> {
+    private elementRef: React.RefObject<HTMLElement>;
+    private observer: typeof ResizeObserver;
+
+    constructor(public props: LayoutProps) {
+        super(props);
+
+        this.handleResize = this.handleResize.bind(this);
+
+        this.state = {
+            maxColumns: MAX_COLUMNS,
+        };
+        this.elementRef = React.createRef();
+    }
+
+    public componentDidMount() {
+        this.observer = new ResizeObserver(this.handleResize);
+        this.observer.observe(this.elementRef.current);
+    }
+
+    public componentWillUnmount() {
+        this.observer.unobserve(this.elementRef.current);
+    }
+
+    public handleResize() {
+        const element = this.elementRef.current;
+        const width = element.getBoundingClientRect().width;
+        const maxColumns = Math.min(
+            MAX_COLUMNS,
+            Math.max(1, Math.floor(width / (MAX_COLUMN_WIDTH * REM_TO_PX)))
+        );
+
+        const columns = this.getColumnCount(maxColumns);
+        if (element.style.getPropertyValue(COLUMN_COUNT) !== String(columns)) {
+            element.style.setProperty(COLUMN_COUNT, String(columns));
```

Comment:
```suggestion
        if (element.style.getPropertyValue(COLUMN_COUNT) !== `${columns}`) {
            element.style.setProperty(COLUMN_COUNT, `${columns}`);
```

I'm not sure it really makes any difference, but the advice at least used to be to avoid the object constructors for types that has literals…

Either way, I kind of think this looks nicer anyway 🙂

Reply:
Errrr… wtf happened to that suggestion???

In case it renders better when reloaded, here's a screenshot:

![image](https://user-images.githubusercontent.com/154725/100339607-5b88cc80-2fda-11eb-859c-efe73863d24e.png)

# 🧐 😂

This is what I _actually_ meant:

![image](https://user-images.githubusercontent.com/154725/100339704-82470300-2fda-11eb-9548-e633788d6c90.png)

Reply:
Update: yeah, it rendered ok after a page reload 😅